app/controllers/trainees/qts_recommendations_controller.rb

```diff
@@ -20,5 +20,9 @@ def create
     def trainee
       @trainee ||= Trainee.from_param(params[:trainee_id])
     end
+
+    def authorize_trainee
+      authorize(trainee)
```

Comment:
This should be a stricter policy, `authorize(trainee)` just checks the user performing the action 'owns' the trainee. However, you should _only_ be able to recommend trainees for QTS who are in the state `trn_received?`.

You could create a method on the policy similar to `def defer?`.

Reply:
Created a `recommend_for_qts?` policy which checks trainee is in `trn_received` state